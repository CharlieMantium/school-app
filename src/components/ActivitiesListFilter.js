import React from 'react';
import Input from './Input';

const ActivitiesListFilter = () => (
  <div>
    <Input type="text" />
  </div>
);

export default ActivitiesListFilter;
