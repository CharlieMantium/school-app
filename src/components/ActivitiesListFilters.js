import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {setTextFilter} from '../actions/filters';

const ActivitiesListFilters = ({ filters: {text}, dispatch }) => (
  <div>
    <input type="text" value={text} onChange={(e) => {
      dispatch(setTextFilter(e.target.value));
    }}/>
  </div>
);

ActivitiesListFilters.propTypes = {
  filters: PropTypes.object,
  text: PropTypes.string,
  dispatch: PropTypes.func
};

const mapStateToProps = (state) => ({
  filters: state.filters
});

export default connect(mapStateToProps)(ActivitiesListFilters);
