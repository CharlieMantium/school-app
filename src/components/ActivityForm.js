import React from 'react';

export default class ActivityForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      classNo: props.activity ? props.activity.classNo : '',
      day: props.activity ? props.activity.day : '',
      name: props.activity ? props.activity.name : '',
      room: props.activity ? props.activity.room : '',
      teacher: props.activity ? props.activity.teacher : '',
      error: ''
    };
  };
  onNameChange = (e) => {
    const name = e.target.value;
    this.setState(() => ({name}));
  };
  onTeacherChange = (e) => {
    const teacher = e.target.value;
    this.setState(() => ({teacher}));
  };
  onRoomChange = (e) => {
    const room = e.target.value;
    this.setState(() => ({room}));
  };
  onClassNoChange = (e) => {
    const classNo = e.target.value;
    if (!classNo || classNo.match(/^[1-9]\d*$/)) {
      this.setState(() => ({classNo}));
    };
  };
  onDayChange = (e) => {
    const day = e.target.value;
    this.setState(() => ({day}));
  };
  onSubmit = (e) => {
    e.preventDefault();
    if (!this.state.name || !this.state.day || !this.state.classNo) {
      this.setState(() => ({error: 'Please provide Activity Name, Day and Class Number!'}))
    } else {
      this.setState(() => ({error: ''}))
      this.props.onSubmit({
        classNo: this.state.classNo,
        day: this.state.day,
        name: this.state.name,
        room: this.state.room,
        teacher: this.state.teacher,
      })
    }
  };
  render() {
    return (
      <div>
        {this.state.error && <p>{this.state.error}</p>}
        <form onSubmit={this.onSubmit}>
          <input
            type="text"
            placeholder="Activity Name"
            autoFocus
            value={this.state.name}
            onChange={this.onNameChange}
          />
          <input
            type="text"
            placeholder="Teacher"
            value={this.state.teacher}
            onChange={this.onTeacherChange}
          />
          <input
            type="text"
            placeholder="Room"
            value={this.state.room}
            onChange={this.onRoomChange}
          />
          <input
            type="number"
            placeholder="Class Number"
            value={this.state.classNo}
            onChange={this.onClassNoChange}
          />
          <select
            value={this.state.day}
            onChange={this.onDayChange}
            required={true}
          >
            <option value=''>Pick a day</option>
            <option value={1}>Monday</option>
            <option value={2}>Tuesday</option>
            <option value={3}>Wednesday</option>
            <option value={4}>Thursday</option>
            <option value={5}>Friday</option>
            <option value={6}>Saturday</option>
            <option value={7}>Sunday</option>
          </select>
          <button>Add Activity</button>
        </form>
      </div>
    )
  }
}
