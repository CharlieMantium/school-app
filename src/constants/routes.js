export const ACTIVITY_PLAN_ROUTE = '/';
export const CREATE_ACTIVITY_ROUTE = '/create';
export const EDIT_ACTIVITY_ID_ROUTE = '/edit/:id';
export const EDIT_ACTIVITY_ROUTE = id => `/edit/${id}`;
