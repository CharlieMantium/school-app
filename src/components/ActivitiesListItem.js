import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { EDIT_ACTIVITY_ROUTE } from '../constants/routes';

const ActivitiesListItem = ({ name, day, classNo, teacher, id }) => (
  <div>
    <Link to={EDIT_ACTIVITY_ROUTE(id)}>
      <h3>{name}</h3>
    </Link>
    <p>{`On ${day} on ${classNo} class with ${teacher}`}</p>
  </div>
);

ActivitiesListItem.propTypes = {
  name: PropTypes.string,
  day: PropTypes.string,
  classNo: PropTypes.string,
  teacher: PropTypes.string,
  id: PropTypes.string
};

export default ActivitiesListItem;
