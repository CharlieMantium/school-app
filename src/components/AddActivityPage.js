import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import AcivityForm from './ActivityForm';
import { addActivity } from '../store/activities/actions';
import { ACTIVITY_PLAN_ROUTE } from '../constants/routes';
import historyPushPropTypeShape from '../prop-types/history';

export const AddActivityPage = ({ onAddActivity, history }) => (
  <div>
    <h1>Add Activity</h1>
    <AcivityForm
      onSubmit={activity => {
        onAddActivity(activity);
        history.push(ACTIVITY_PLAN_ROUTE);
      }}
    />
  </div>
);

AddActivityPage.propTypes = {
  onAddActivity: PropTypes.func.isRequired,
  history: PropTypes.shape(historyPushPropTypeShape).isRequired,
};

const mapDispatchToProps = {
  onAddActivity: addActivity,
};

export default connect(
  null,
  mapDispatchToProps,
)(AddActivityPage);

// TODO: Ask Lukasz about null in export

// TODO: handle absolute imports
