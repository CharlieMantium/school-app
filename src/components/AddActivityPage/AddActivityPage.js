import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { FormattedMessage } from 'react-intl';

import { startAddActivity } from 'store/activities/actions';
import { ACTIVITY_PLAN_ROUTE } from 'constants/routes';
import historyPushPropTypeShape from 'prop-types/history';
import { Heading } from 'styles/elements/Heading';
import { ActivityFormWrapper } from 'styles/elements/ActivityFormWrapper';

import AcivityForm from '../ActivityForm';

const AddActivityPage = ({ onStartAddActivity, history }) => (
  <ActivityFormWrapper>
    <Heading as="h1">
      <FormattedMessage id="form.heading.addActivityHeading" />
    </Heading>
    <AcivityForm
      onSubmit={activity => {
        onStartAddActivity(activity);
        history.push(ACTIVITY_PLAN_ROUTE);
      }}
    />
  </ActivityFormWrapper>
);

AddActivityPage.propTypes = {
  onStartAddActivity: PropTypes.func.isRequired,
  history: PropTypes.shape(historyPushPropTypeShape).isRequired,
};

const mapDispatchToProps = {
  onStartAddActivity: startAddActivity,
};

export { AddActivityPage as AddActivityPageUnwrapped };
export default connect(
  null,
  mapDispatchToProps,
)(AddActivityPage);
