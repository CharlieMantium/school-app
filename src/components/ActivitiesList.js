import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import ActivitiesListItem from './ActivitiesListItem';
import selectActivities from '../store/activities/selectors/activities';

const ActivitiesList = ({ activities }) => (
  <div>
    <h1>Activities List</h1>
    {activities.map((activity) => (
      <ActivitiesListItem {...activity} key={activity.id}/>
    ))}
  </div>
);

ActivitiesList.propTypes = {
  activities: PropTypes.array
};

const mapStateToProps = (state) => ({
  activities: selectActivities(state.activities, state.filters)
});

export default connect(mapStateToProps)(ActivitiesList);
