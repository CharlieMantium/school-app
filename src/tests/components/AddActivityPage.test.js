import React from 'react';
import { shallow } from 'enzyme';

import { AddActivityPage } from '../../components/AddActivityPage';
import { ACTIVITY_PLAN_ROUTE } from '../../constants/routes';
import testState from '../fixtures/state';

let onAddActivitySpy;
let historySpy;
let wrapper;

beforeEach(() => {
  onAddActivitySpy = jest.fn();
  historySpy = { push: jest.fn() };
  wrapper = shallow(<AddActivityPage onAddActivity={onAddActivitySpy} history={historySpy} />);
});

describe('AddActivityPage', () => {
  it('should render AddActivityPage correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle onSubmit', () => {
    wrapper.find('ActivityForm').prop('onSubmit')(testState.activities.items[1]);
    expect(historySpy.push).toHaveBeenLastCalledWith(ACTIVITY_PLAN_ROUTE);
    expect(onAddActivitySpy).toHaveBeenLastCalledWith(testState.activities.items[1]);
  });
});
