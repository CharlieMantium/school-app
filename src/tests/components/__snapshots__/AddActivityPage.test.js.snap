// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddActivityPage should render AddActivityPage correctly 1`] = `
<div>
  <h1>
    Add Activity
  </h1>
  <ActivityForm
    activity={
      Object {
        "classNo": "",
        "day": "",
        "name": "",
        "room": "",
        "teacher": "",
      }
    }
    onSubmit={[Function]}
  />
</div>
`;
